src/wrappers/PhylorefWrapper.js

```diff
@@ -643,48 +660,108 @@ class PhylorefWrapper {
-    } else {
-      // Step 1. Construct an expression for all internal specifiers.
-      let expressionsForInternals;
+    } else if (externalSpecifiers.length > 0) {
+      // If the phyloreference has at least one external specifier, we
+      // can provide a simplified expression for the internal specifier,
+      // in the form:
+      //  phyloref:includes_TU some [internal1] and
+      //  phyloref:includes_TU some [internal2] and ...
+      // To which we can then add the external specifiers.
       if (internalSpecifiers.length === 1) {
-        expressionsForInternals = [
+        logicalExpressions = PhylorefWrapper.createClassExpressionsForExternals(
+          phylorefAsJSONLD,
           PhylorefWrapper.getIncludesRestrictionForTU(internalSpecifiers[0]),
-        ];
+          externalSpecifiers,
+          []
+        );
       } else {
-        expressionsForInternals = PhylorefWrapper.createClassExpressionsForInternals(
-          phylorefAsJSONLD, internalSpecifiers, []
+        const expressionForInternals = {
+          '@type': 'owl:Class',
+          intersectionOf: internalSpecifiers.map(PhylorefWrapper.getIncludesRestrictionForTU),
+        };
+
+        logicalExpressions = PhylorefWrapper.createClassExpressionsForExternals(
+          phylorefAsJSONLD, expressionForInternals, externalSpecifiers, []
         );
       }
+    } else {
+      // We only have internal specifiers. We therefore need to use the algorithm in
+      // PhylorefWrapper.createClassExpressionsForInternals() to create this expression.
+      logicalExpressions = PhylorefWrapper.createClassExpressionsForInternals(
+        phylorefAsJSONLD, internalSpecifiers, []
+      );
+    }
 
-      if (externalSpecifiers.length === 0) {
-        // If we don't have external specifiers, we can just use the expression
-        // for the internal specifier.
-        phylorefAsJSONLD.equivalentClass = expressionsForInternals;
-      } else {
-        // Step 2. Create alternate class expressions for external specifiers.
-        phylorefAsJSONLD.equivalentClass = expressionsForInternals.map(
-          exprForInternal => PhylorefWrapper.createClassExpressionsForExternals(
-            phylorefAsJSONLD, exprForInternal, externalSpecifiers, []
-          )
-        ).reduce((acc, val) => acc.concat(val), []);
+    // If we have a single logical expression, we set that as an equivalentClass
+    // expression. If we have more than one, we produce multiple component classes
+    // to represent it.
+    if (logicalExpressions.length === 0) {
+      // This is fine, as long as there is an explanation in
+      // phyloref.malformedPhyloreference explaining why no logical expressions
+      // could be generated. Otherwise, throw an error.
+      if (!has(phylorefAsJSONLD, 'malformedPhyloreference')) {
+        throw new Error(`No logical expressions generated for phyloref ${this.label}, but no explanation was provided either.`);
```

Comment:
If I read that as something spilled onto the terminal, I'd think the program is trying to blame me, the user. Either say who didn't provide this, or simply say _Unexpectedly no logical expression generated for ..._ Again, small nitpick.

Reply:
Good catch! Fixed in e080def67a93.